### PR TITLE
Fortran docs

### PR DESCRIPTION
**Pull request by billschereriii**
De-macro procedure definitions to make source code more compatible with Doxygen

### CODE REVIEW

File: src/fortran/client.F90 [hunk 1]
```diff
@@ -408,6 +408,11 @@ end function poll_key
 !> Put a tensor whose Fortran type is the equivalent 'int8' C-type
 function put_tensor_i8(self, name, data, dims) result(code)
   integer(kind=c_int8_t), dimension(..), target, intent(in) :: data !< Data to be sent
+  class(client_type),                    intent(in) :: self !< Fortran SLIC client
```
Comment (ashao) [inline src/fortran/client.F90:411]: SLIC/SILC is a holdover from the early days of SmarTsim. Maybe rename to "An initialized SmartRedis Fortran client"

File: tests/cpp/test_cpp_client.py [hunk 2]
```diff
@@ -62,7 +62,7 @@ def execute_cmd(cmd_list):
     proc = Popen(
         cmd_list, stderr=PIPE, stdout=PIPE, stdin=PIPE, cwd=run_path)
     try:
-        out, err = proc.communicate(timeout=120)
+        out, err = proc.communicate(timeout=180)
```
Comment (ashao) [inline tests/cpp/test_cpp_client.py:65]: Any reason in particular why this was increased?

**Review by ashao — CHANGES_REQUESTED** [review]
Other than a minor change in the description and a question about the timeout in the CPP tests, this is pretty much good to go. Thanks @billschereriii

File: tests/cpp/test_cpp_client.py (hunk 2 above)
Comment (ashao) [reply tests/cpp/test_cpp_client.py:65]: Also, we should make sure we make these magic numbers more obvious.

File: src/fortran/client.F90 (hunk 1 above)
Comment (billschereriii) [reply src/fortran/client.F90:411]: Sure. Didn't even look at it when I copy/pasted it

File: tests/cpp/test_cpp_client.py (hunk 2 above)
Comment (billschereriii) [reply tests/cpp/test_cpp_client.py:65]: The CPP unit tests are timing out occasionally in the checkin, so I bumped it to 3 minutes from 2, Should really have done this in a separate PR but I was being lazy.  I agree in general about magic numbers, but this is the driver for our test cases, so not production code

Comment (ashao) [reply tests/cpp/test_cpp_client.py:65]: Agreed. Mainly just a note to myself to think about how to consolidate these kinds of things.

**Review by ashao — APPROVED** [review]
Thanks @billschereriii. Feel free to merge

**Pull request merged**